### ISSUE TITLE
how about the training and inference speed

### ISSUE THREAD

**Issue by GallonDeng** [HaozhiQi/ISONet#1]
very interesting work. In the paper, there are no discusssions about the differences on the training and inference speed compared to standard resnet. Are there any differences? Thanks

**Comment by HaozhiQi** [comment]
Thanks for your interest.

For the 34 layer network, the training time is **11mins** per epoch for ISONet and **9.5mins** for ResNet with 8 Titan RTX. The main slowdown is due to the implementation of orthogonal regularization. Currently, I took the most naive way: [sequentially compute the regularization loss described in (13)/(14)](https://github.com/HaozhiQi/ISONet/blob/master/isonet/models/isonet.py#L399). But it can be improved by concatenating multiple convolutional kernels (within in the same stage) and computing the loss by group convolution.

For inference, note that no regularization is applied. Therefore ISONet / R-ISONet has almost the same (if not faster, due to removing BatchNorm) inference speed compared to ResNet.
